### ISSUE TITLE
HttpBindings don't work for minimal app without UseUrls

### ISSUE THREAD

**Issue by 0x53A** [CoreWCF/CoreWCF#119]
If you have a minimal asp.net core application without either

```c#
.UseUrls(...)
```

or

```c#
.UseKestrel(options => { options.ListenLocalhost(<port>);}
``` 

then at initialization time, ``app.ServerFeatures.Get<IServerAddressesFeature>().Addresses`` is empty.

This will cause CoreWCF to throw later:

```
System.InvalidOperationException: 'Could not find a base address that matches scheme http for the endpoint with binding NetHttpBinding. Registered base address schemes are [].'
```

-----------

I found two possible solutions:

1) add ``http://localhost:5000`` as default

In src\CoreWCF.Http\src\CoreWCF\Channels\ServiceModelHttpMiddleware.cs ``BuildBranch`` just add 

```c#
            if (!serviceBuilder.BaseAddresses.Any())
            {
                // https://github.com/aspnet/Announcements/issues/224
                // The WebHost will no longer add the default server address of http://localhost:5000 to the IServerAddressesFeature when none is specified.
                serviceBuilder.BaseAddresses.Add(new Uri("http://localhost:5000"));
                serviceBuilder.BaseAddresses.Add(new Uri("https://localhost:5001"));
            }
```

But this is a wild guess that may or may not be correct.

<br />
<br />
  
-----------
  

2) delay CoreWCF initialization until server has been started

After kestrel has been started, ``IServerAddressesFeature.Addresses`` WILL be populated correctly.

In the same file, ``private readonly RequestDelegate _branch;`` could be changed to ``private readonly Lazy<RequestDelegate> _branch;`` and it seems to work.

------------

Note, there is another issue, where if you call just ``UseNetTcp`` without ``UseUrls``, then it will listen __only__ on the nettcp port, and no longer on 5000/5001.

**Comment by mconnew** [comment]
I've got a fix in the works. The problem is CoreWCF needs to know the URL at service setup as it is used as a parameter for getting tokens from a token provider. But ASP.NET Core doesn't populate the features until after Kestrel is running. I have a fix which lazily creates the service, but I'm getting an intermittent 404 response when testing. Once I've tracked down why I sometimes get a 404, I'll push the fix.
